### PR TITLE
OGM-1429 Support run java code for embedded mode

### PR DESCRIPTION
**Pull request by ntviet18**

### CODE REVIEW

**Comment by ntviet18** [comment]
@schernolyas please review

File: infinispan-embedded/src/main/java/org/hibernate/ogm/datastore/infinispan/InfinispanDialect.java [hunk 1]
```diff
@@ -62,7 +78,10 @@
  * @author Emmanuel Bernard
  * @author Fabio Massimo Ercoli
  */
-public class InfinispanDialect<EK,AK,ISK> extends BaseGridDialect {
+public class InfinispanDialect<EK,AK,ISK> extends BaseGridDialect
+		implements StoredProcedureAwareGridDialect {
```
Comment (schernolyas) [inline infinispan-embedded/src/main/java/org/hibernate/ogm/datastore/infinispan/InfinispanDialect.java:82]: it must be in one row ( see other dialects as example)

File: infinispan-embedded/src/main/java/org/hibernate/ogm/datastore/infinispan/InfinispanDialect.java [hunk 2]
```diff
@@ -228,6 +247,93 @@ public void forEachTuple( ModelConsumer consumer, TupleTypeContext tupleTypeCont
 		return (KeyProvider<EK, AK, ISK>) provider.getKeyProvider();
 	}
 
+	@Override
+	public ClosableIterator<Tuple> callStoredProcedure( String storedProcedureName, ProcedureQueryParameters queryParameters, TupleContext tupleContext ) {
+		EmbeddedCacheManager embeddedCacheManager = getCacheManager().getCacheManager();
+		Cache<String, String> cache = embeddedCacheManager.getCache( "___stored_procedures", true );
+		String className = cache.getOrDefault( storedProcedureName, storedProcedureName );
```
Comment (schernolyas) [inline infinispan-embedded/src/main/java/org/hibernate/ogm/datastore/infinispan/InfinispanDialect.java:254]: I  supposed something like https://apacheignite.readme.io/docs/deployment-spi#section-example .
I mean you can use full class name as stored procedure name

File: infinispan-embedded/src/main/java/org/hibernate/ogm/datastore/infinispan/util/ReflectionHelper.java [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+	 *
+	 * @param obj object for introspection.
+	 *
+	 * @return a map containing object's field values.
+	 *
+	 * @throws IntrospectionException if an exception occurs during introspection
+	 * @throws InvocationTargetException if property getter throws an exception
+	 * @throws IllegalAccessException if property getter is inaccessible
+	 */
+	public static Map<String, Object> introspect(Object obj)
```
Comment (schernolyas) [inline infinispan-embedded/src/main/java/org/hibernate/ogm/datastore/infinispan/util/ReflectionHelper.java:38]: format to one row

File: infinispan-embedded/src/test/java/org/hibernate/ogm/datastore/infinispan/test/storedprocedures/InfinispanStoredProcedureCallExceptionsTest.java [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+	public void tearDown() {
+		em.close();
+	}
+
+	@Test
+	public void testExceptionWhenInvalidProcedureName() throws Exception {
+		thrown.expect( PersistenceException.class );
+		thrown.expectMessage( "org.hibernate.HibernateException: OGM001113: Cannot instantiate stored procedure 'invalidProcedureName' with resolved name 'invalidProcedureName'." );
+		StoredProcedureQuery storedProcedureQuery = em.createStoredProcedureQuery( "invalidProcedureName", Car.class );
+		storedProcedureQuery.registerStoredProcedureParameter( "param", Integer.class, ParameterMode.IN );
```
Comment (schernolyas) [inline infinispan-embedded/src/test/java/org/hibernate/ogm/datastore/infinispan/test/storedprocedures/InfinispanStoredProcedureCallExceptionsTest.java:52]: You offer use named parameters. It is ok. But .. please throw exception if I try use positional parmeters

File: infinispan-embedded/src/test/java/org/hibernate/ogm/datastore/infinispan/test/storedprocedures/ResultSetProcedure.java [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+		this.id = id;
+	}
+
+	public void setTitle(String title) {
+		this.title = title;
+	}
+
+	@Override
+	public Car call() {
+		return new Car( id, title );
```
Comment (schernolyas) [inline infinispan-embedded/src/test/java/org/hibernate/ogm/datastore/infinispan/test/storedprocedures/ResultSetProcedure.java:33]: Where is result set? Return please collection of Car

**Comment by ntviet18** [comment]
Please review https://github.com/epam/hibernate-ogm/pull/42 instead

**Pull request closed without merging**